Sources/Sentry/SentryScope.m

```diff
@@ -63,6 +65,10 @@
 
 @property (atomic, strong) NSMutableArray<SentryAttachment *> *attachmentArray;
 
+@property (nullable, atomic, strong) id<SentrySpan> currentTransaction;
+
+@property (nullable, atomic, strong) NSString *fallbackTransactionName;
```

Comment:
M: what's the inspiration for using `fallback`? it's an old feature, it's not really a fallback

Reply:
with a proper performance API, maybe we don't even need to support this old feature, @bruno-garcia @philipphofmann or at least not do in this PR, because right now it sets a `transactionName` but its not read from anywhere, this would go to `event.transaction` during `applyScope`

Reply:
This behavior exists in both .NET and Java, thats why I implemented it here. 

TransactionName is used in serialization in case a transaction is not provided.

Reply:
@brustolin yep but that's not the point, transactionName now is being set to the Scope but it's never set to SentryEvent, so it's doing nothing.

Reply:
Understood. If is an old feature lets remove it and start cocoa free of deprecated stuff.

Reply:
I'm a bit lost now. `transaction` (a string) is part of the protocol forever and it goes with all events. We need to keep that.
It became an issue when we started setting a transaction (instance) to the scope since the property itself conflicted. 

It's still possible to have a transaction (name) on the scope withut having performance monitoring turned on at all. And `captureMessage` and friends should continue to send that transaction name. Is this the behavior here?

Reply:
@bruno-garcia iOS didn't have the `event.transaction` feature which is a String and it's in the Scope, I suggested not adding here because it's not the goal of this PR, if we want to have that feature still, we'd make a new PR

Reply:
Yes that makes sense.
Maybe we don't even add this (transactionName) then.